### DISCUSSION TITLE
Implement fulltext search with SQLModel and PostgresSQL

### DISCUSSION THREAD

**Discussion by ppvan** [fastapi/sqlmodel#769] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional

from sqlmodel import Field, SQLModel, create_engine


class Hero(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str
    secret_name: str
    age: Optional[int] = None


sqlite_file_name = "database.db"
sqlite_url = f"sqlite:///{sqlite_file_name}"

engine = create_engine(sqlite_url, echo=True)

SQLModel.metadata.create_all(engine)
```

### Description

Hi, I read all the docs and user guide but can't figure out how to implement fulltext search in sqlmodel. For example, search all hero contains name "spider". I'd like to perform some analayis like ascii folding if possible.

Thanks in advanced.

### Operating System

Linux

### SQLModel Version

0.0.14

### Python Version

Python 3.11.6

**Comment by machinehead** [comment]
I was wondering about the same thing and just found this SO answer: https://stackoverflow.com/a/9552871/12423765

Translated it into roughly this sqlmodel code below, and it seems to work!

(I omitted all the irrelevant details of my code from this, please let me know if I failed to include something important)

```python
from typing import Sequence, Self
from sqlalchemy.dialects.postgresql.ext import to_tsquery
from sqlmodel import Field, SQLModel, select, col
from sqlmodel.ext.asyncio.session import AsyncSession

class Note(SQLModel, table=True):
    content: str = Field(nullable=False)

    @classmethod
    async def search_by_text(cls, session: AsyncSession, query: str) -> Sequence[Self]:
        return (
            await session.exec(
                select(cls).filter(col(cls.content).op("@@")(to_tsquery(query)))
            )
        ).all()

```
